File: Java/DesignPatterns/src/main/java/designPatterns/structural/composite/MainComposite.java
package designPatterns.structural.composite;

public class MainComposite {
  public static void main(String[] args) {
    Leaf.A leafA = new Leaf.A("leafA", 100);
    Leaf.B leafB = new Leaf.B("leafB", 101, true);
    Composite composite = new Composite();
    composite.addComponent(leafA);
    composite.addComponent(leafB);

    System.out.println(composite.show());
  }
}

File: Kotlin/DesignPatterns/src/main/kotlin/structural/composite/MainComposite.kt
package structural.composite

fun main() {
  val leafA = Leaf.A("leafA", 100)
  val leafB = Leaf.B("leafB", 101, true)
  val composite = Composite()
  composite.addComponent(leafA)
  composite.addComponent(leafB)

  print(composite.show())
}